server.js

```diff
@@ -1,23 +1,180 @@
 const express = require("express");
 const cors = require("cors");
+const bodyParser = require("body-parser");
 
 const app = express();
 
 app.use(cors());
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+
+//handling CORS
+// app.use((request ,response, next)=>{
+//   response.header('Access-Control-Allow-Origin','*');
+//   response.header('Access-Control-Allow-Headers', '*');
+//   if(request.method === 'OPTIONS'){
+//     response.header('Access-Control-Allow-Methods','PUT, POST, PATCH, DELETE, GET');
+//     return response.status(200).json({});
+//   }
+//   next();
+// })
+
+//handling 404 error
+// app.use((request, response, next) =>{
+//   const error = new Error("Not Found");
+//   error.status = 404;
+//   next(error);
+// })
 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
-  text: "Welcome to CYF chat system!",
+  text: "Welcome to CYF chat system!"
 };
 
 //This array is our "data store".
 //We will start with one message in the array.
 //Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+// const messages = [welcomeMessage];
+const messages =[
+  {
+      id: 0,
+      from: "Bart",
+      text: "Welcome to CYF chat system!"
+    },
+    {
+      id: 1,
+      from: "Bart",
+      text: "I hope you enjoy this course"
+    },
+    {id:2,
+    from:"farhana",
+    text:"hello"},
+    {id:3,
+    from:"humail",
+    text:"bye"},
+    {
+      id: 4,
+      from: "Bart",
+      text: "Welcome to CYF chat system!"
+    },
+    {
+      id: 5,
+      from: "Bart",
+      text: "I hope you enjoy this course"
+    },
+    {id:6,
+    from:"farhana",
+    text:"hello"},
+    {id:7,
+    from:"humail",
+    text:"bye"},
+    {
+      id: 8,
+      from: "Bart",
+      text: "Welcome to CYF chat system!"
+    },
+    {
+      id: 9,
+      from: "Bart",
+      text: "I hope you enjoy this course"
+    },
+    {id:10,
+    from:"farhana",
+    text:"hello"},
+    {id:11,
+    from:"humail",
+    text:"bye"},
+    {
+      id:12,
+      from: "dania",
+      text: "call back"
+    },
+    {
+      id:13,
+      from:"fatima",
+      text:"sorry busy rn!"
+    },
+
+]
+    
 
-app.get("/", function (request, response) {
+
+app.get("/", function(request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+//Post message
+app.post("/messages", function(request, response) {
+  if (request.body.from === "" || request.body.text === "") {
+    response.status(400).json({
```

Comment:
Nice use of a status 400. Sometimes it can be difficult to choose the right status codes, but I think you're correct here in classifying an incomplete form as a bad request 😃